crates/autopilot/src/driver_model.rs

```diff
@@ -150,6 +154,30 @@ pub mod solve {
         pub submission_address: H160,
     }
 
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub enum FeePolicy {
+        /// Applies to limit orders only.
+        /// This fee should be taken if the solver provided good enough solution
+        /// that even after the surplus fee is taken, there is still more
+        /// surplus left above whatever the user expects [order limit price
+        /// or best quote, whichever is better for the user].
+        QuoteDeviation {
+            /// Percentage of the order's `available surplus` should be taken as
+            /// a protocol fee.
+            ///
+            /// `Available surplus` is the difference between the executed_price
+            /// (adjusted by surplus_fee) and the closer of the two: order
+            /// limit_price or best_quote. For out-of-market limit orders,
+            /// order limit price is closer to the executed price. For
+            /// in-market limit orders, best quote is closer to the executed
+            /// price.
```

Comment:
Let's not define a new keyword here ("available surplus"). We have called this price improvement in the past.

```suggestion
            /// Factor of price improvement the protocol charges as a fee.
            /// Price improvement is the difference between executed price and
            /// limit price or quoted price (whichever is better)
            /// 
            /// E.g. if a user received 2000USDC for 1ETH while having been quoted 1990USDC,
            /// their price improvement is 10USDC. A factor of 0.5 requires the solver to pay
            /// 5USDC to the protocol for settling this order.
```

Reply:
added